File: Backend/VestTour.Domain/Entities/Fabric.cs
﻿using System;
using System.Collections.Generic;

namespace VestTour.Domain.Entities;

public partial class Fabric
{
    public int FabricId { get; set; }

    public string? FabricName { get; set; }

    public decimal? Price { get; set; }

    public string? Description { get; set; }

    public string? ImageUrl { get; set; }

    public string? Tag { get; set; }

    public string? Status { get; set; }

    public virtual ICollection<Product> Products { get; set; } = new List<Product>();
}

File: Mobile/mobileapp/lib/models/fabric.dart
class Fabric {
  int? fabricID;
  String? fabricName;
  double? price;
  String? description;
  String? imageUrl;
  int? tag;

  Fabric(
      {this.fabricID,
        this.fabricName,
        this.price,
        this.description,
        this.imageUrl,
        this.tag});

  Fabric.fromJson(Map<String, dynamic> json) {
    fabricID = json['fabricID'];
    fabricName = json['fabricName'];
    price = json['price'];
    description = json['description'];
    imageUrl = json['imageUrl'];
    tag = json['tag'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['fabricID'] = this.fabricID;
    data['fabricName'] = this.fabricName;
    data['price'] = this.price;
    data['description'] = this.description;
    data['imageUrl'] = this.imageUrl;
    data['tag'] = this.tag;
    return data;
  }
}
